### DISCUSSION TITLE
How can you reset the state of a device in HomeKit?

### DISCUSSION THREAD

**Discussion by iKrushYou** [HomeSpan/HomeSpan#924] [General]
I updated a characteristic, specifically I changed the range of it. I don't see the changes reflected in the Home app. I know that I can remove the device and re-pair it and that will solve it, but I already have this device added to a lot of scenes / automations and I'd prefer not to do that. is it possible to clear a state to have it reset?

**Comment by HomeSpan** [comment]
When you updated the range in the sketch HomeSpan would have automatically increased with configuration number, so HomeKit sees a new value.  The Home App is supposed to take this into account and reset values, but I have found it is not always as reliable as it should be.   You may want to try changing the range again to see if it takes effect on the Home App.

On a related note, recently a user found that setting the range for some Characteristics had no effect on newer versions of the Home App, though it did in older versions.  According to the original HAP-R2 doc, range changes only affect "int" and "float" Characteristics though in practice they worked with others as well.   What Characteristic are you trying to change?

**Reply by iKrushYou** [reply]
I was trying to update the Fan:RotationSpeed characteristic. I was switching between 0,100 and 0,5

```
        fanRotationSpeed = new Characteristic::RotationSpeed();
        fanRotationSpeed->setRange(0, 5);
```

but simply changing it did not update it in the home app. I'm on 18.1 
I ended up resetting the device from the CLI, removing it from Home and adding it back
